server/src/http/unleash_client.rs

```diff
@@ -97,30 +100,49 @@ impl UnleashClient {
         Ok(Self {
             urls: UnleashUrls::from_str(server_url)?,
             backing_client: new_reqwest_client(instance_id),
+            custom_headers: Default::default(),
         })
     }
 
     fn client_features_req(&self, req: ClientFeaturesRequest) -> RequestBuilder {
         let client_req = self
             .backing_client
             .get(self.urls.client_features_url.to_string())
-            .header(reqwest::header::AUTHORIZATION, req.api_key);
+            .headers(self.header_map(Some(req.api_key)));
         if let Some(tag) = req.etag {
-            client_req.header(reqwest::header::IF_NONE_MATCH, tag.to_string())
+            client_req.header(header::IF_NONE_MATCH, tag.to_string())
         } else {
             client_req
         }
     }
 
-    #[instrument]
+    fn header_map(&self, api_key: Option<String>) -> HeaderMap {
+        let mut header_map = HeaderMap::new();
+        if let Some(key) = api_key {
+            header_map.insert(header::AUTHORIZATION, key.parse().unwrap());
```

Comment:
Minor detail but these unwraps can fail if someone passes goop into the headers. Agree that a panic is a good choice here, but an `expect` over an `unwrap`, might be nice for understanding what the user did wrong here. WDYT?

Reply:
Agreed. will fix